File: backend/test_connection.py
from pymongo import MongoClient
import os
from dotenv import load_dotenv

load_dotenv()

MONGO_URI = os.getenv("MONGO_URI")

try:
    client = MongoClient(MONGO_URI)
    client.admin.command('ping')  # check connection
    db = client["unirank"]
    print("[OK] MongoDB Connected Successfully")
except Exception as e:
    print("[ERROR] MongoDB Connection Failed:", e)

File: backend/src/scripts/testConnection.js
/**
 * scripts/testConnection.js
 * Tests MongoDB connection.
 */

import dotenv from 'dotenv';
import mongoose from 'mongoose';

dotenv.config();

const uri = process.env.MONGO_URI || 'mongodb://localhost:27017/unirank';

try {
  await mongoose.connect(uri);
  console.log('[OK] MongoDB Connected Successfully');
  await mongoose.disconnect();
} catch (err) {
  console.error('[ERROR] MongoDB Connection Failed:', err.message);
}
